lib/msal-node/src/cache/serializer/Serializer.ts

```diff
@@ -34,10 +46,20 @@ export class Serializer {
      * Serialize IdTokens
      * @param idTCache
      */
-    static serializeIdTokens(idTCache: IdTokenCache): StringDict{
-        const idTokens: StringDict = {};
-        Object.keys(idTCache).map(function (key) {
-            idTokens[key] = JSON.stringify(EntitySerializer.mapIdTokenKeys(idTCache, key));
+    static serializeIdTokens(
+        idTCache: IdTokenCache
+    ): Record<string, SerializedIdTokenEntity> {
+        const idTokens: Record<string, SerializedIdTokenEntity> = {};
+        Object.keys(idTCache).map(function(key) {
```

Comment:
Every place where you instantiate a new object, map over an existing object and add values onto the new object can be written to use reduce instead as shown above.

Reply:
Please check the above comment. I will have to test once and will update in a follow up PR.